omnigibson/prims/cloth_prim.py

```diff
@@ -136,6 +179,361 @@ def _post_load(self):
+                            biggest_face_num = face_num
+                    new_ms = pymeshlab.MeshSet()
+                    for split_mesh in ms:
+                        if split_mesh.face_number() == biggest_face_num:
+                            new_ms.add_mesh(split_mesh)
+                    ms = new_ms
+
+                avg_edge_percentage_mismatch = abs(
+                    1.0 - particle_distance / ms.get_geometric_measures()["avg_edge_length"]
+                )
+            else:
+                # Terminate anyways, but don't fail
+                log.warning("The generated cloth may not have evenly distributed particles.")
+
+            # Check if we have too many vertices
+            cm = ms.current_mesh()
+            if cm.vertex_number() > m.MAX_CLOTH_PARTICLES:
+                # We have too many vertices, so we will re-mesh again
+                particle_distance *= math.sqrt(2)  # halve the number of vertices
+                log.warning(
+                    f"Too many vertices ({cm.vertex_number()})! Re-meshing with particle distance {particle_distance}..."
+                )
+            else:
+                break
+        else:
+            raise ValueError(f"Could not remesh with less than MAX_CLOTH_PARTICLES ({m.MAX_CLOTH_PARTICLES}) vertices!")
+
+        # Re-write data to @mesh_prim
+        new_faces = cm.face_matrix()
+        new_vertices = cm.vertex_matrix()
+        new_normals = cm.vertex_normal_matrix()
+        texcoord = cm.wedge_tex_coord_matrix() if has_uv_mapping else None
+        tm = trimesh.Trimesh(
+            vertices=new_vertices,
+            faces=new_faces,
+            vertex_normals=new_normals,
+        )
+        # Apply the inverse of the world transform to get the mesh back into its local frame
+        tm.apply_transform(th.linalg.inv_ex(scaled_world_transform).inverse)
+
+        # Update the mesh prim to store the new information. First update the non-configuration-
+        # dependent fields
+        face_vertex_counts = th.tensor([len(face) for face in tm.faces], dtype=int).cpu().numpy()
+        self.prim.GetAttribute("faceVertexCounts").Set(face_vertex_counts)
+        self.prim.GetAttribute("faceVertexIndices").Set(tm.faces.flatten())
+        self.prim.GetAttribute("normals").Set(lazy.pxr.Vt.Vec3fArray.FromNumpy(tm.vertex_normals))
+        if has_uv_mapping:
+            self.prim.GetAttribute("primvars:st").Set(lazy.pxr.Vt.Vec2fArray.FromNumpy(texcoord))
+
+        # Remove the properties for all configurations
+        for config in CLOTH_CONFIGURATIONS:
+            attr_name = f"points_{config}"
+            if self.prim.HasAttribute(attr_name):
+                self.prim.RemoveProperty(attr_name)
+
+        # Then update the configuration-dependent fields
+        self.save_configuration("default", th.tensor(tm.vertices, dtype=th.float32))
+
+        # Then update the points to the default configuration
+        self.reset_points_to_configuration("default")
```

Comment:
Is this needed?

Reply:
This is what actually sets the `points` attribute to also use the remeshed configuration (note that we don't set it elsewhere here). I guess it's kind of a sanitycheck too.